src/transformers/models/ibert/utils/quant_modules.py

```diff
@@ -0,0 +1,538 @@
+
+            # Initialization
+            if torch.eq(self.x_min, self.x_max).all():
+                self.x_min = self.x_min + x_min
+                self.x_max = self.x_max + x_max
+
+            # exponential moving average (EMA)
+            # use momentum to prevent the quantized values change greatly every iteration
+            elif self.act_range_momentum == -1:
+                self.x_min = torch.min(self.x_min, x_min)
+                self.x_max = torch.max(self.x_max, x_max)
+            else:
+                self.x_min = self.x_min * self.act_range_momentum +\
+                        x_min * (1 - self.act_range_momentum)
+                self.x_max = self.x_max * self.act_range_momentum +\
+                        x_max * (1 - self.act_range_momentum)
+
+        if not self.quant_mode:
+            return x_act, None
+
+        x_min = self.x_min if specified_min is None else specified_min
+        x_max = self.x_max if specified_max is None else specified_max
+
+        self.act_scaling_factor = symmetric_linear_quantization_params(
+            self.activation_bit, x_min, x_max, 
+            per_channel=self.per_channel)
+
+        if pre_act_scaling_factor is None:
+            # this is for the input quantization 
+            quant_act_int = self.act_function(x, self.activation_bit, \
+                    self.percentile, self.act_scaling_factor)
+        else:
+            quant_act_int = fixedpoint_mul.apply(
+                    x, pre_act_scaling_factor, 
+                    self.activation_bit, self.quant_mode, 
+                    self.act_scaling_factor, 
+                    identity, identity_scaling_factor)
+
+        correct_output_scale = self.act_scaling_factor.view(-1)
+
+        return quant_act_int * correct_output_scale, self.act_scaling_factor
+
+
+class QuantLinear(Module):
+    """
+    Class to quantize weights of given Linear layer
+    
+    Parameters:
+    ----------
+    weight_bit : int
+        Bitwidth for quantized weights.
+    bias_bit : int, default None
+        Bitwidth for quantized bias.
+    per_channel : bool, default False
+        Whether to use channel-wise quantization.
+    quant_mode : bool, default False
+        The mode for quantization. True for quantization.
+    """
+    def __init__(self,
+                 weight_bit,
+                 bias_bit=None,
+                 per_channel=False,
+                 quant_mode=False):
+        super(QuantLinear, self).__init__()
+        self.weight_bit = weight_bit
+        self.quant_mode = quant_mode
+        self.per_channel = per_channel
+        self.bias_bit = bias_bit
+        self.quantize_bias = (False if bias_bit is None else True)
+        self.quant_mode = quant_mode
+        self.percentile_mode = False
+
+        if not self.quant_mode:
+            self.weight_function = None
+        else:
+            self.weight_function = SymmetricQuantFunction.apply
```

Comment:
Think it's cleaner to just have `self.weight_function = SymmetricQuantFunction.apply` -> it doesn't make too much sense to set a function to `None` if it's not used IMO